python/tvm/datatype.py

```diff
@@ -0,0 +1,73 @@
+"""Custom datatype functionality"""
+from __future__ import absolute_import as _abs
+
+from ._ffi.function import register_func as _register_func
+from . import make as _make
+from .api import convert
+from .expr import Call as _Call, Cast as _Cast
+from ._ffi.runtime_ctypes import TVMType as _TVMType
+from . import _api_internal
+
+
+def _datatype_register_op(extern_func_name,
```

Comment:
Expose lower so that users *can* use it if they want but are not forced to; this function should just become a thin wrapper at that point. See register_intrin_rule.